theforeman.org/pipelines/release/source/foreman-maintain.groovy

```diff
@@ -0,0 +1,78 @@
+pipeline {
+    agent any
+
+    options {
+        timestamps()
+        timeout(time: 2, unit: 'HOURS')
+        ansiColor('xterm')
+        buildDiscarder(logRotator(daysToKeepStr: '7'))
+    }
+
+    stages {
+        stage('Test') {
+            matrix {
+                agent any
+                axes {
+                    axis {
+                        name 'ruby'
+                        values '2.7', '3.0'
+                    }
+                }
+                stages {
+                    stage("clone repository") {
+                        steps {
+                            git url: git_url, branch: git_ref
+                        }
+                    }
+                    stage("unit test") {
+                        steps {
+                            configureRVM(env.ruby, project_name)
+                            withRVM(["bundle install --with=development --jobs=5 --retry=5"], env.ruby, project_name)
+                            archiveArtifacts(artifacts: 'Gemfile.lock')
+                            withRVM(["bundle exec rake jenkins:unit --trace"], env.ruby, project_name)
+                        }
+                    }
+                }
+                post {
+                    cleanup {
+                        cleanupRVM(env.ruby, project_name)
+                        junit(testResults: 'jenkins/reports/unit/*.xml')
+                        deleteDir()
+                    }
+                }
+            }
+        }
+        stage('Build and Archive Source') {
+            steps {
+                dir(project_name) {
+                    git url: git_url, branch: git_ref
+
+                    script {
+                        archive_git_hash()
+                    }
+                }
+                script {
+                    sourcefile_paths = generate_sourcefiles(project_name: project_name, source_type: source_type)
```

Comment:
It won't work, because this function does a `dir`:

https://github.com/theforeman/jenkins-jobs/blob/536e1138951959c6c658aa4cafb3d05245dfa8e7/theforeman.org/pipelines/lib/nightly_packaging.groovy#L7

Reply:
Ah yes. That's confusing.